cpp/src/parquet/encoding.cc

```diff
@@ -3177,7 +3179,9 @@ class RleBooleanDecoder : public DecoderImpl, virtual public BooleanDecoder {
         PARQUET_THROW_NOT_OK(
             out->AppendValues(values.begin(), values.begin() + current_batch_size));
         num_values -= current_batch_size;
-        current_index_in_batch = 0;
+        // set current_index_in_batch to current_batch_size means
+        // the whole batch is totally consumed.
+        current_index_in_batch = current_batch_size;
```

Comment:
This is another bug I found by testing. "Release" will not trouble by this, but `DCHECK` might failed in `next_boolean_batch`

Reply:
What's the point of `current_index_in_batch` exactly? It always ends up having the same value.

Reply:
It's only useful in per-value next when `null_count != 0`. `next_boolean_batch` will check all values in this batch is consumed (.i.e current_index_in_batch == batch_size). When `null_count != 0`, it's useless actually, but it uses `next_boolean_batch` here.